### ISSUE TITLE
Recruitment candidate user account(s)

### ISSUE THREAD

**Issue by MandarinDuck** [ministryofjustice/analytical-platform#6184]
### Describe the feature request.

One or more user accounts on the Analytical Platform for use by non-MoJ users for the purposes of technical assessments used in recruitment.
The requirements would include:

- The ability to run VSCode, Jupyter Lab, and R-Studio as does any typical AP user.
- For admins to be readily and frequently able to clear the NAS storage for these accounts (so a fresh image is available to each candidate without any persistence of files between interviews). During interview periods this would be daily if not several times per day, depending on how many 'guest' accounts existed for use.
- Having read access to only specific named repositories in the MoJ Github organisations (ie. unable to see organisation-level repos, and only be able to see repos specifically created for recruitment exercises - or public repos).
- [Maybe] for recruitment managers to have access to these accounts, or otherwise have a means to create these images when designing the technical assessment exercises. In the interests of being user-friendly, it might just be that assessors could log-in and pre-clone a repo for the candidate, but really the candidate could be talked through this process at the start of an assessment too, so this is not an essential requirement.

IAM access to S3 appears to already be adequate for these accounts (ie. there is no access by default), but it would be worth confirming information security at some stage.

### Describe the context.

During recruitment for Data Science and Data Engineering roles there is a need to administer technical tests for assessing candidates. This typically means providing access to computing resources to test the candidates' skills are appropriate for the role.

Currently we use free online computing resources such as Google Colaboratory or Binder, but this presents several difficulties:

- As free services these are unreliable and may subject users to timeouts mid-assessment (Binder in particular does so after 10 minutes of inactivity).
- Google Colab requires a Google account to use, which some applicants may not have, or not wish to use in these circumstances. Requiring them to sign up to a Google account to apply also means they must agree to Google's terms of service, which could be considered an unreasonable requirement to be considered for employment.
- Any data used in the assessment must be made public (eg. via Github or Google Drive). Synthetically generated data is sometimes preferable to open source data as we can be sure candidates have not used it before and can have a full understanding of its content (the patterns and relationships within it).
- We cannot curate the environment and ensure relevant tools are installed in advance, so candidates lose time waiting for their chosen software to install, or worse, since they might be unfamiliar platforms, find themselves unexpectedly unable to use their preferred tools during the assessment.
- MoJ recruitment assessments are temporarily public and/or being exposed to third-party organisations

### Value / Purpose

In summary:

- Provides a fairer and more stable environment for recruitment
- Enables higher-quality recruitment exercises to be set
- Resolves some information security concerns

Guest accounts would effectively resolve all the concerns in 'context' above, and would also provide an environment that accurately represents the working environment of a data professional in the MoJ, so candidates would be better informed regarding the nature of role they have applied for.

### User Types

MoJ job applicants, Recruitment managers

**Comment by MandarinDuck** [comment]
We'd also want to create a separate terms-of-use agreement for these accounts.

**Comment by simon-pope** [comment]
Request forum: Ask understood, will take to Refinement

**Comment by MandarinDuck** [comment]
Has there been any advancement on this? We are currently in the middle of a recruitment campaign using Google Colab and its negative aspects are quite tangible.

**Comment by andreassoteriadesmoj** [comment]
Indeed, it would be great to know if there has been any progress on this. It would be unreasonable to expect candidates to agree to Google's terms of service or not be able to interview for the role in future recruitment rounds.

**Comment by andreassoteriadesmoj** [comment]
Any news on this please?

**Comment by andreassoteriadesmoj** [comment]
Any news on this please?

**Comment by gfowler-moj** [comment]
@MandarinDuck @andreassoteriadesmoj Apologies for the delayed response. Unfortunately this isn't something that we have the capacity to support at the moment. 

While I appreciate that this would be a great way to test **_potential_** data analysts/engineers, I'd also be a little reluctant to develop specific features within AP that would be accessible by non-MoJ staff anyway (this would bring some additional risks with it).
